argilla-server/src/argilla_server/api/schemas/v1/records.py

```diff
@@ -205,15 +200,20 @@ def prevent_nan_values(cls, metadata: Optional[Dict[str, Any]]) -> Optional[Dict
 
         return {k: v for k, v in metadata.items() if v == v}  # By definition, NaN != NaN
 
+    def is_set(self, attribute: str) -> bool:
+        return attribute in self.__fields_set__
 
-class RecordUpdateWithId(RecordUpdate):
-    id: UUID
+    def has_changes(self) -> bool:
+        return self.dict(exclude_unset=True) != {}
```

Comment:
Should we use `model_dump` instead?

```suggestion
        return self.model_dump(exclude_unset=True) != {}
```